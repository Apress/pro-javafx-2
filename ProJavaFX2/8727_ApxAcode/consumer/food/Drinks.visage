/*
 * @subtest
 */

// Drinks.visage
package food;

public class Coffee {
  public var brand: String;
}

public class Tea {
  public var kind: String;
}
package var drinksOffered:Integer;

public function getCoffee():Coffee {
  drinksOffered++;
  Coffee { brand: "Folgers" }
}
public function getTea():Tea {
  drinksOffered++;
  Tea { kind: "Iced" }
}
public function numberOfDrinksOffered() {
  drinksOffered
}
