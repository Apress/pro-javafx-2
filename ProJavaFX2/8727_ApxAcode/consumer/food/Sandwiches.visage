/*
 * @subtest
 */

// Sandwiches.visage
package food;

public class Club {
  public-init var kind: String;
  public-read var price: Number;
}

public function calculatePrice(club: Club) {
  if (club.kind == "Roast Beef") {
    club.price = 7.99;
  } else if (club.kind == "Chicken") {
    club.price = 6.99;
  }
}
