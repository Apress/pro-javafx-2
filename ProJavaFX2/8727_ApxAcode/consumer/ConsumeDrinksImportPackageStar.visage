/*
 * @test
 * @compile ../food/Drinks.visage
 * @package consumer
 * @run
 */

// ConsumeDrinksImportPackageStar.visage
package consumer;

import food.*;

var coffee = Drinks.getCoffee();
println("Good coffee.  It's {coffee.brand}.");
var tea = Drinks.getTea();
println("Good tea.  It's {tea.kind} tea.");
println("Number of drinks offered = {Drinks.numberOfDrinksOffered()}.");
