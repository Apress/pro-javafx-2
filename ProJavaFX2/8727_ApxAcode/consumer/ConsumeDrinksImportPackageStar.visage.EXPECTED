Good coffee.  It's Folgers.
Good tea.  It's Iced tea.
Number of drinks offered = 2.
